index.js

```diff
@@ -35,23 +35,28 @@ module.exports = function (src, dest, opts, cb) {
 				return;
 			}
 
-			var read = fs.createReadStream(src);
-			var write = fs.createWriteStream(dest);
-
-			read.on('error', done);
-			write.on('error', done);
-			write.on('close', function () {
-				fs.lstat(src, function (err, stats) {
-					if (err) {
-						done(err);
-						return;
-					}
-
+			fs.lstat(src, function (err, stats) {
+				if (err) {
+					done(err);
+					return;
+				}
+
+				if (stats.isDirectory()) {
```

Comment:
Checking if a file is not a directory before writing to it is vulnerable to race conditions: another process may remove the file and create a directory between the calls to stats.isDirectory() and fs.createReadStream(src)…